plugins/module_utils/vm_snapshot.py

```diff
@@ -158,10 +191,32 @@ def to_ansible(self) -> TypedVMSnapshotToAnsible:
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, VMSnapshot):
             return NotImplemented
+
+        check_vm = True  # it will be True if self.vm == None
+        if self.vm != {}:
+            vm_sorted_block_devices = [
+                dict(sorted(bd.items(), key=lambda item: item[0]))  # type: ignore
+                for bd in self.vm["block_devices"]
+            ]
+            other_sorted_block_devices = [
+                dict(sorted(bd.items(), key=lambda item: item[0]))  # type: ignore
+                for bd in other.vm["block_devices"]
+            ]
```

Comment:
Yes, I am trying to sort the ``block_devices`` dict by its key ``uuid``, not the whole ``vm``. With this, I ensure that when testing this class, I don't get a false negative because the ``block_devices`` keys are differently arranged. This is only needed for ``block_devices``.

Reply:
This was resolved. :)